src/NumberGuessingGame.jsx

```diff
@@ -13,65 +13,42 @@ function getRandomNumber() {
 
 const MAX_ATTEMPTS = 5;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
+function NumberGuessingGame() {
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
```

Comment:
To prevent repeating that computation of the `getRandomNumber` function, you could pass an initializer function to `setState`.

```suggestion
  const [numberToGuess, setNumberToGuess] = useState(() => getRandomNumber());
```

---

The code below has the same effect but is a bit more optimized:

```suggestion
  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber);
```